src/uipath/_services/attachments_service.py

```diff
@@ -0,0 +1,780 @@
+                )
+                print(f"Uploaded attachment with key: {attachment_key}")
+            ```
+        """
+        spec = self._create_attachment_and_retrieve_upload_uri_spec(
+            name=name,
+            folder_key=folder_key,
+            folder_path=folder_path,
+        )
+
+        result = (
+            await self.request_async(
+                spec.method,
+                url=spec.endpoint,
+                params=spec.params,
+                headers=spec.headers,
+                json=spec.json,
+            )
+        ).json()
+
+        # Get the ID from the response and convert to UUID
+        attachment_key = uuid.UUID(result["Id"])
+
+        upload_uri = result["BlobFileAccess"]["Uri"]
+        headers = {
+            key: value
+            for key, value in zip(
+                result["BlobFileAccess"]["Headers"]["Keys"], result["BlobFileAccess"]["Headers"]["Values"], strict=False
+            )
+        }
+
+        # Convert string to bytes if needed
+        if isinstance(content, str):
+            content = content.encode("utf-8")
+
+        if result["BlobFileAccess"]["RequiresAuth"]:
+            await self.request_async(
+                "PUT", upload_uri, headers=headers, content=content
+            )
+        else:
+            request("PUT", upload_uri, headers=headers, content=content)
+            
+        return attachment_key
+
+    @traced(name="attachments_list", run_type="uipath")
+    def list(
+        self,
+        *,
+        folder_key: Optional[str] = None,
+        folder_path: Optional[str] = None,
+    ) -> List[Attachment]:
+        """List all attachments.
+
+        This method retrieves a list of all attachments from UiPath.
+
+        Args:
+            folder_key (Optional[str]): The key of the folder. Override the default one set in the SDK config.
+            folder_path (Optional[str]): The path of the folder. Override the default one set in the SDK config.
+
+        Returns:
+            List[Attachment]: List of attachments.
+
+        Examples:
+            ```python
+            from uipath import UiPath
+
+            client = UiPath()
+
+            attachments = client.attachments.list()
+            for attachment in attachments:
+                print(attachment.name)
+            ```
+        """
+        spec = self._retrieve_attachments_spec(
+            folder_key=folder_key,
+            folder_path=folder_path,
+        )
+
+        response = self.request(
+            spec.method,
+            url=spec.endpoint,
+            params=spec.params,
+            headers=spec.headers,
+        ).json()
+
+        # Ensure required fields exist in the response items
+        config_info = self._get_tenant_info()
+
+        return [
+            self._ensure_attachment_fields(item, config_info)
+            for item in response["value"]
+        ]
+
+    @traced(name="attachments_list_async", run_type="uipath")
```

Comment:
do we actually want to add a method to the SDK that returns all the attachments in the specified folder without pagination?